SpatialGDK/Documentation/content/singleton-actors.md

```diff
@@ -1,55 +1,84 @@
 
-Due to server-workers spawning their own instances of each Singleton Actor, proper replication and authority management of Singleton Actors becomes a bit tricky. To solve this issue, we have introduced the concept of a Global State Manager (GSM) to enable proper replication of Singleton Actors. The GSM solves the problem of replicating Singleton Actors by only allowing the server-worker with [authority]({{urlRoot}}/content/glossary#authority) over the GSM to execute the initial replication of these Actors. All other server-workers will then link their local Singleton Actors to their respective SpatialOS entity. Because of this, you must update your snapshot whenever adding a new Singleton Actor to your project.
+Before you begin to use Singleton Actors across your game world, review the following high-level procedure. See each section for more details:
 
-## Setting up Singleton Actors
+1. [Define Singleton Actors](#defining-singleton-actors): Manually tag an Unreal C++ class or an Unreal Blueprint class with the relevant attributes to be a Public Singleton Actor or Private Singleton Actor.
+2. [Spawn Singleton Actors](#spawning-singleton-actors): Write code to spawn Singleton Actors on server-worker instances at runtime. 
 
-To set up Singleton Actors for your project, you need to:
+In addition, learn about how SpatialOS can [manage Singleton Actors](#managing-singleton-actors) to ensure that there is only ever one instance of an entity that represents a Singleton Actor, and to ensure data is replicated properly. 
 
-1. Register Singleton Actors by tagging them with the `SpatialType=Singleton` class attribute. If you wish to make them Private Singletons, tag them with the additional `ServerOnly` class attribute.
+## Defining Singleton Actors
 
-The code snippet below shows how to tag a native C++ class with the appropriate Public Singleton identifiers.
+Before you begin, ensure that you have an Actor, which can be either an [Unreal C++ Class](https://docs.unrealengine.com/en-us/Programming/Development/ManagingGameCode/CppClassWizard) or an [Unreal Blueprint Class](https://docs.unrealengine.com/en-US/Engine/Blueprints/UserGuide/Types/ClassBlueprint).
 
-```
-UCLASS(SpatialType=Singleton)
-class TESTSUITE_API AExampleGameGameState : public AGameStateBase
-{
-  GENERATED_BODY()
-  ...
-}
-```
+### Unreal C++ Class
 
-The code snippet below shows how to tag a native C++ class with the appropriate Private Singleton identifiers.
+1. Open your C++ Class in any text editor.
+2. Tag the C++ Class with the relevant attribute(s), depending on what type of Singleton Actor you want it to be:
```

Comment:
Should this number indexing all be `1`s?

Reply:
Good point, yes it should! (@eddyrainy if you use "1" for every numbered step, this gets translated into the correct number in the actual docs. It means you can add more steps in the middle without having to re-number everything.)